katcp/kattypes.py

```diff
@@ -305,8 +305,12 @@ class Discrete(Str):
     def __init__(self, values, case_insensitive=False, **kwargs):
         super(Discrete, self).__init__(**kwargs)
         self._case_insensitive = case_insensitive
+        if not isinstance(values, list):
```

Comment:
Do we really need to be this strict?  A tuple or similar sequence would also be fine any.  The casting to `list(values)` would work with all of those.  Even with a generator, after the change on line 313

I.e., follow the duck-typing approach that Python encourages (at least before mypy... 🙂 ) .

Reply:
Mmmhh yeah a bit too strict from my end, but as per the docstring https://github.com/ska-sa/katcp-python/blob/653d69e7d23c977bbb8d263380ab0d1aec5134a0/katcp/kattypes.py#L297 values should a list.

Will update the code. Thanks.

Reply:
Docstrings can be fixed :-) 

> I.e., follow the duck-typing approach that Python encourages (at least before mypy... 

Mypy supports duck-typing by means of "protocols" e.g. one could annotate this argument as `Iterable[str]`.

Reply:
Please update the PR title and description as well.  The main reason for doing this was to handle generators.